tools/crates-validator/src/profiles/parser/mod.rs

```diff
@@ -0,0 +1,200 @@
+use std::fmt::Display;
+
+use anyhow::Result;
+
+use super::Combine;
+use super::Profile;
+use super::ProfileSettings;
+use super::RiscZeroRepo;
+
+mod utils;
+mod batch;
+mod profiles;
+
+pub fn get_profiles(path: impl AsRef<str> + Display) -> Result<Vec<Profile>> {
+    // Read the whole configuration file
+    let content = utils::read_profile(path)?;
+
+    todo!()
+    // Parse batch configurations
+    // let batch_profiles = batch::parse(&content)?;
+
+    // Parse individual crate configurations
+    // let individual_profiles = profiles::parse(&content)?;
+
+    // Combine the two, overwriting batch profiles with individual profiles if
+    // crates are defined in both
+    // let profiles = utils::combine_profiles(batch_profiles, individual_profiles)?;
+
+    // Ok(profiles)
+}
```

Comment:
Need to leverage `parse` functions defined in submodules `batch` and `profiles`.

Reply:
Maybe also split the function into reading and parsing, and also export `parse_profile` as public?